### ISSUE TITLE
Configuration syntax errors produce unreadable errors

### ISSUE THREAD

**Issue by stelcheck** [mage/mage#150]
Example, with a YAML config file:

```> crash@0.0.1 develop /tmp/crash
>   echo 'Run MAGE in development mode' &&             npm run mage

Run MAGE in development mode

> crash@0.0.1 mage /tmp/crash
>      echo 'Run the mage binary'  &&                     mage

Run the mage binary

Failed to load MAGE application:

Error
    at generateError (/tmp/crash/node_modules/js-yaml/lib/js-yaml/loader.js:162:10)
    at throwError (/tmp/crash/node_modules/js-yaml/lib/js-yaml/loader.js:168:9)
    at readDocument (/tmp/crash/node_modules/js-yaml/lib/js-yaml/loader.js:1513:5)
    at loadDocuments (/tmp/crash/node_modules/js-yaml/lib/js-yaml/loader.js:1549:5)
    at load (/tmp/crash/node_modules/js-yaml/lib/js-yaml/loader.js:1566:19)
    at Object.safeLoad (/tmp/crash/node_modules/js-yaml/lib/js-yaml/loader.js:1584:10)
    at loadConfigFile (/tmp/crash/node_modules/mage/lib/config/config.js:48:17)
    at loadConfig (/tmp/crash/node_modules/mage/lib/config/config.js:155:16)
    at Object.exports.initialize (/tmp/crash/node_modules/mage/lib/config/config.js:393:18)
    at Object.<anonymous> (/tmp/crash/node_modules/mage/lib/config/index.js:6:34)

Make sure to properly set the value of "main" in your package.json,
and that your code is valid.
```

This does not allow the developer to know:

  1. Which file needs fixing
  2. What caused the error (what line, ideally)

cc @ronkorving @christopherobin

**Comment by stelcheck** [comment]
Addendum: JSON format has some code to deal with that, but the output is not exactly helpful I think.

```
Failed to load MAGE application:

Error: Parse error on line 1:
# crash - developmen
^
Expecting 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '[', got 'undefined'
    at Object.parseError (/tmp/crash/node_modules/jsonlint/lib/jsonlint.js:55:11)
    at Object.parse (/tmp/crash/node_modules/jsonlint/lib/jsonlint.js:132:22)
    at Object.exports.parse (/tmp/crash/node_modules/jsonlint/lib/jsonlint.js:417:53)
    at loadConfigFile (/tmp/crash/node_modules/mage/lib/config/config.js:60:14)
    at loadConfig (/tmp/crash/node_modules/mage/lib/config/config.js:155:16)
    at Object.exports.initialize (/tmp/crash/node_modules/mage/lib/config/config.js:393:18)
    at Object.<anonymous> (/tmp/crash/node_modules/mage/lib/config/index.js:6:34)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)

Make sure to properly set the value of "main" in your package.json,
and that your code is valid.
```

It is *better*, but if we are to fix this for all supported format we'd benefit from reviewing how JSON configuration syntax errors are presented.

We will also need to make sure #147 has proper code to deal with syntax errors as well.

**Comment by ronkorving** [comment]
I think jsonlint output is really good actually. You should try it on some more json-esque data. The output is great.

Maybe there are some good yaml linters out there?

**Comment by stelcheck** [comment]
I agree the output is ok, but we don't know which file it comes from, for
instance.

**Comment by stelcheck** [comment]
This can greatly be improved by making output errors from `bin/mage` more readable. #147 will take care of improving this.

**Comment by stelcheck** [comment]
Closed by #115
